edx_notifications/stores/sql/store_provider.py

```diff
@@ -184,6 +185,9 @@ def _get_prepaged_notifications(self, user_id, filters=None, options=None):
         if type_name:
             query = query.filter(msg__msg_type=type_name)
 
+        if start_date and end_date:
+            query = query.filter(created__gte=start_date, created__lte=end_date)
+
```

Comment:
Sorry - hate to say this, but we should be able to filter either by start_date *or* end_date *or* both. Something like:

if start_date:
   query = query.filter(created__gte=start_date)

if end_date:
   query=query.filter(created__lte=end_date)

And maybe add some additional tests to:

filter only by start_date
filter only by end_date

Thanks.